lib/client/sso/redirector.go

```diff
@@ -269,7 +286,17 @@ func (rd *Redirector) clickableURL(redirectURL string) string {
 	// short path is a link-shortener style URL
 	// that will redirect to the Teleport-Proxy supplied address
 	rd.mux.HandleFunc(shortPath, func(w http.ResponseWriter, r *http.Request) {
-		http.Redirect(w, r, redirectURL, http.StatusFound)
+		if postForm != "" {
+			form, err := base64.StdEncoding.DecodeString(postForm)
+			if err != nil {
+				http.Redirect(w, r, "/", trace.ErrorToCode(err))
```

Comment:
What does the user see when that happens?

Reply:
The http handler set up by tsh serves raw HTTP response. In this case, user would see HTTP error code converted from the err. 
I have now updated that to actually also include error message that is responded proper with [http.Error](https://github.com/gravitational/teleport/pull/54605/commits/47cc8b5db940cdbc13b01de57ce6dfb35a92885a#diff-fe14f312a603291e87bc4972f34773a298661b587bebc4a3d6672dd73d3c6f57R292)